lib/device.ex

```diff
@@ -0,0 +1,149 @@
+defmodule Onvif.Device do
+  alias Onvif.Device
+  alias Onvif.Discovery.Probe
+
+  @enforce_keys [:username, :password, :ip]
+
+  defstruct @enforce_keys ++
+              [
+                :auth_type,
+                :scopes,
+                :manufacturer,
+                :model,
+                :firmware_version,
+                :serial_number,
+                :hardware_id,
+                :datetime,
+                :ntp,
+                :media_service_path,
+                current_diff: 0,
+                port: 80,
+                supports_media2?: false,
+                device_service_path: "/onvif/device_service",
+                is_nvr?: false
+              ]
+
+  @spec init(Onvif.Discovery.Probe.t(), String.t(), String.t(), boolean, boolean) ::
+          {:error, any()}
+          | %Device{}
+  def init(%Probe{} = probe_match, username, password, use_https? \\ false, use_ipv6? \\ false) do
+    with {:ok, device} <-
+           device_from_probe_match(probe_match, username, password, use_https?, use_ipv6?),
+         {:ok, device_with_datetime} <- get_date_time(device),
+         {:ok, updated_device} <- guess_auth(device_with_datetime) do
+      updated_device
+      |> get_services()
+      |> set_media_service_path()
+    end
+  end
+
+  defp device_from_probe_match(%Probe{} = probe_match, username, password, use_https?, use_ipv6?) do
+    scheme = if use_https?, do: "https", else: "http"
+
+    uri =
+      probe_match.address
+      |> get_address(use_ipv6?)
+      |> URI.parse()
+      |> Map.put(:scheme, scheme)
+      |> Map.put(:userinfo, "#{username}:#{password}")
+
+    is_nvr? = if "tds:Device" in probe_match.types, do: false, else: true
```

Comment:
I don't think this is correct. I think `if "onvif://www.onvif.org/type/Net_Video_Recorder" in probe_match.scopes` would be more accurate (but may miss some if the manufacturer doesn't return accurate scopes).

Reply:
Since the scope `onvif://www.onvif.org/type/Net_Video_Recorder` is also not guaranteed by the  ONVIF spec, removed the `is_nvr?` scope from the library. Only profile, location, hardware, and name are guaranteed.

![image (1)](https://github.com/hammeraj/onvif/assets/9105503/da772ac0-6ee9-490a-942d-46ddc4849cc4)